instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -189,8 +189,14 @@ def _instrumented_lambda_handler_call(
 
         # See more:
         # https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html
+        # https://docs.aws.amazon.com/lambda/latest/dg/with-sns.html
+        # https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html
+        # https://docs.aws.amazon.com/lambda/latest/dg/with-ddb.html
+        span_kind = SpanKind.SERVER
```

Comment:
is there any change when this default is incorrect?

Reply:
Right now the default is implicitly set to null which doesn't allow the code to even proceed past the usage of span_kind downstream.  Seems like the original intent was to look for SQS and call it a consumer if it didn't exist in the map.  I'm interested to hear @NathanielRN 's opinion though.

Reply:
Yes this was an oversight on my part, appreciate your deep dive into the code @brett-bim! In general we assumed that we wanted all spans to be Server spans.

(This is important for AWS X-Ray because only Server spans get converted to segments).